storage-proto/src/convert.rs

```diff
@@ -343,7 +343,9 @@ impl From<generated::Message> for VersionedMessage {
             .into_iter()
             .map(|key| Pubkey::try_from(key).unwrap())
             .collect();
-        let recent_blockhash = Hash::new(&value.recent_blockhash);
+        let recent_blockhash = <[u8; HASH_BYTES]>::try_from(value.recent_blockhash)
+            .map(Hash::new_from_array)
+            .unwrap();
```

Comment:
A lot of callers need to bring in `HASH_BYTES` and do this conversion -- for easier use, how about adding a `TryFrom<&[u8]> for Hash` implementation, similar to what we have with `Pubkey`?

https://github.com/anza-xyz/agave/blob/8b720f78c3f78ff4b8d32a33f0a41999e645e1b7/sdk/pubkey/src/lib.rs#L258-L265